### PR TITLE
Feature/remove opencart

### PR DESCRIPTION
**Pull request by vitormattos**
**What?** Remove OpenCart from composer.json and put in RoboFile
**Why?** The OpenCart from composer has many dependencies in composer, this dependencies affects others packages

### CODE REVIEW

**Comment by michelpl** [comment]
Use commit emojis.

File: RoboFile.php [hunk 1]
```diff
@@ -28,47 +30,159 @@ public function __construct()
...
+        $url = 'https://github.com/opencart/opencart/releases/download/' . getenv('OPENCART_VERSION') . '/' . $filename;
+        $headers = get_headers($url, 1);
+        $filesize = $headers['Content-Length'];
+        if (!file_exists($filename) || filesize($filename) != $filesize) {
+            $remote = fopen($url, 'r');
+            if ($remote === false) {
+                throw new Exception('Failed to access '.$url);
+            }
+            $local = fopen($filename, 'w');
+            if ($remote === false) {
```
Comment (felipebool) [inline RoboFile.php]: I think you wanted to test another variable in this if

File: RoboFile.php [hunk 2]
```diff
@@ -28,47 +30,159 @@ public function __construct()
...
-            } catch (PDOException $e) {
-                $this->say("<error> Database error: " . $e->getMessage());
-            }
+    private function dropIfExistsAndCreateDatabase()
+    {
+        if (!getenv('REGENERATE_OPENCART_DIR')) {
+            return;
+        }
+        try {
+            $conn = new PDO('mysql:host=' . $this->opencart_config['db_hostname'], $this->opencart_config['db_username'], $this->opencart_config['db_password']);
```
Comment (felipebool) [inline RoboFile.php]: Break this line

File: RoboFile.php [hunk 3]
```diff
@@ -28,47 +30,159 @@ public function __construct()
...
         }
+        $this->taskDeleteDir(getenv('OC_ROOT'))->run();
+        $filename = getenv('OPENCART_VERSION') . '-OpenCart.zip';
+        $this->downloadOpenCart($filename);
+        $this->extractOpenCart($filename);
+    }
 
+    private function downloadOpenCart(string $filename)
+    {
+        $url = 'https://github.com/opencart/opencart/releases/download/' . getenv('OPENCART_VERSION') . '/' . $filename;
```
Comment (michelpl) [inline RoboFile.php]: Break line

Comment (felipebool) [inline RoboFile.php]: Think about this "Blank lines MAY be added to improve readability and to indicate related blocks of code." for next pull requests

File: RoboFile.php [hunk 4]
```diff
@@ -28,47 +30,159 @@ public function __construct()
...
+            ->to(
+                "error_reporting(E_ALL);\r\n" .
+                "define('DIR_STORAGE', '".__DIR__."/system/storage/');"
+            )
+            ->run();
 
-            if (!is_dir('.modman')) {
-                $this->taskExec('vendor/bin/modman init ' . getenv('OC_ROOT'))->run();
-                $this->taskExec('ln -sf "$(pwd)" .modman/')->run();
-            }
```
Comment (michelpl) [inline RoboFile.php]: Ver o que aconteceu aqui.

File: RoboFile.php [hunk 5]
```diff
@@ -28,47 +30,159 @@ public function __construct()
...
+        $this->taskConcat([
+            getenv('OC_ROOT') . 'tmp_setup.php',
+            getenv('OC_ROOT').'install/cli_install.php'
+        ])
+        ->to(getenv('OC_ROOT').'install/cli_install.php')
+        ->run();
+
+        $install = $this->taskExec('php')->arg(getenv('OC_ROOT').'install/cli_install.php')->arg('install');
```
Comment (felipebool) [inline RoboFile.php]: Add one blank space before and after a '.' when concatenating strings.
https://github.com/php-fig-rectified/fig-rectified-standards/blob/master/PSR-2-R-coding-style-guide-additions.md

Comment (felipebool) [reply RoboFile.php]: This problem is all over this pull request

File: RoboFile.php (hunk 4 above)
Comment (vitormattos) [reply RoboFile.php]: This block is moved to new method installModuleByModman

File: RoboFile.php (hunk 5 above)
Comment (vitormattos) [reply RoboFile.php]: This repository don't is the official PHP-FIG repository. In official PSR-2, don't has this exigence.

**Pull request merged**